packages/grid/x-data-grid-pro/src/components/GridRowReorderCell.tsx

```diff
@@ -39,11 +39,15 @@ const GridRowReorderCell = (params: GridRenderCellParams) => {
   const cellValue = params.row.__reorder__ || params.id;
 
   // TODO: remove sortModel and treeDepth checks once row reorder is compatible
-  const isDraggable =
-    !!(rootProps as DataGridProProcessedProps).rowReordering &&
-    !sortModel.length &&
-    treeDepth === 1 &&
-    Object.keys(editRowsState).length === 0;
+  const isDraggable = React.useMemo(
+    () =>
+      !!(rootProps as DataGridProProcessedProps).rowReordering &&
+      !sortModel.length &&
+      treeDepth === 1 &&
+      Object.keys(editRowsState).length === 0,
+    // eslint-disable-next-line
+    [rootProps.rowReordering, sortModel, treeDepth, editRowsState],
```

Comment:
Yes, as long as you import from the `useGridRootProps` of your current package, you should have the interface of the props of the current package.
But if you import it from `@mui/x-data-grid` (which you should not), it will fail.

So probably a legacy thing we forgot to remove during the introduction of the row reorder.